src/passes/OptimizeInstructions.cpp

```diff
@@ -606,6 +606,18 @@ struct OptimizeInstructions
             }
           }
         }
+        if (binary->op == DivFloat32) {
+          float c = right->value.getf32();
+          if (Bits::isPowerOf2Float(c)) {
+            return optimizePowerOf2FDiv(binary, c);
+          }
+        }
+        if (binary->op == DivFloat64) {
```

Comment:
these two could be combined into a single block. A single can check both opcodes, and then `right->value.getFloat()` (which like `getInteger()`, gets a value of either 32 or 64 bits, as 64 bits - safe here as all floats fit precisely as doubles).

Reply:
Also we would then only need `isPowerOfTwoFloat` on doubles, and not on floats, so it removes code there too

Reply:
Just realized it's not valid. Some tests failed if force conversion to doubles. Like:
```
f32(x) / 5.877471754111438e-39
```
becomes
```
f32(x) * 1701411834604692317316873e14
```
that's because dynamic range of float and doubles quite different.

Reply:
Hmm, actually this refactoring also icover more valid cases